openmdao/components/balance_comp.py

```diff
@@ -170,11 +170,12 @@ def apply_nonlinear(self, inputs, outputs, residuals):
 
             if options['normalize']:
                 # Indices where the rhs is near zero or not near zero
-                idxs_nz = np.where(cs_safe.abs(rhs) < 2)[0]
-                idxs_nnz = np.where(cs_safe.abs(rhs) >= 2)[0]
+                idxs_nz = np.where(cs_safe.abs(rhs) < 2)
+                idxs_nnz = np.where(cs_safe.abs(rhs) >= 2)
 
                 # Compute scaling factors
                 # scale factor that normalizes by the rhs, except near 0
+                self._scale_factor = np.ones((rhs.shape))
```

Comment:
If we're reallocating _scale_factor here on each pass through, then it should just be a local variable and not an attribute of the class.  Move it up out of the `if` clause here and remove the else-clause so that it's correct when `normalize` is `False`.